File: backend/HabitHub.Api/Models/Session.cs
using System.Xml.Serialization;
using Microsoft.AspNetCore.Mvc.ViewFeatures;
using HabitHub.Api.Enums;

namespace HabitHub.Api.Models;

public class Session
{
    public Guid SessionId { get; set; }

    public Guid MemberId { get; set; }
    public Member Member { get; set; } = null!;

    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
    public DateTime LastActiveAt { get; set; }
    public DateTime ExpiresAt { get; set; }
    public SessionState State { get; set; } = SessionState.Active;
    public string Device { get; set; } = string.Empty;
    public string IPAddress { get; set; } = string.Empty;
}

File: frontend/app/dto/Session.ts
export enum SessionState {
  Active,
  Invalidated,
  Expired
}

export type Session = {
  sessionId: string;
  memberId: string;
  createdAt: string;
  lastActiveAt: string;
  expiresAt: string;
  device?: string;
  ipAddress?: string;
  state: SessionState;
};